rllib/algorithms/appo/torch/appo_torch_learner.py

```diff
@@ -0,0 +1,187 @@
+from typing import Any, Dict, Mapping
+
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.algorithms.appo.appo_learner import (
+    AppoLearner,
+    LEARNER_RESULTS_CURR_KL_COEFF_KEY,
+    LEARNER_RESULTS_KL_KEY,
+    OLD_ACTION_DIST_KEY,
+)
+from ray.rllib.algorithms.impala.torch.vtrace_torch_v2 import (
+    make_time_major,
+    vtrace_torch,
+)
+from ray.rllib.core.learner.learner import POLICY_LOSS_KEY, VF_LOSS_KEY, ENTROPY_KEY
+from ray.rllib.core.learner.torch.torch_learner import TorchLearner
+from ray.rllib.core.rl_module.marl_module import ModuleID
+from ray.rllib.utils.annotations import override
+from ray.rllib.utils.framework import try_import_torch
+from ray.rllib.utils.typing import TensorType
+
+torch, nn = try_import_torch()
+
+
+class APPOTorchLearner(TorchLearner, AppoLearner):
+    """Implements APPO loss / update logic on top of ImpalaTorchLearner."""
+
+    def __init__(self, *args, **kwargs):
+        TorchLearner.__init__(self, *args, **kwargs)
+        AppoLearner.__init__(self, *args, **kwargs)
+
+    @override(TorchLearner)
+    def compute_loss_per_module(
+        self, module_id: str, batch: SampleBatch, fwd_out: Mapping[str, TensorType]
+    ) -> TensorType:
+
+        values = fwd_out[SampleBatch.VF_PREDS]
+        target_policy_dist = fwd_out[SampleBatch.ACTION_DIST]
+        old_target_policy_dist = fwd_out[OLD_ACTION_DIST_KEY]
+        old_target_policy_actions_logp = old_target_policy_dist.logp(
+            batch[SampleBatch.ACTIONS]
+        )
+        behaviour_actions_logp = batch[SampleBatch.ACTION_LOGP]
+        target_actions_logp = target_policy_dist.logp(batch[SampleBatch.ACTIONS])
+
+        behaviour_actions_logp_time_major = make_time_major(
+            behaviour_actions_logp,
+            trajectory_len=self.hps.rollout_frag_or_episode_len,
+            recurrent_seq_len=self.hps.recurrent_seq_len,
+            drop_last=self.hps.vtrace_drop_last_ts,
```

Comment:
should we just stop using this in the new appo impl/ we could start by removing it here in this file and the tf learner file. then later we can introduce a pr to remove it from the make_time_major code.

Reply:
Actually, that's even better. Will do.

Reply:
done, only valid now for `enable_learner_api=False`.